character-sfx.js

```diff
@@ -0,0 +1,107 @@
+/* this is the character sfx implementation.
+it is used for footsteps sfx etc */
+
+import * as THREE from 'three';
+import gameManager from './game.js';
+import cameraManager from './camera-manager.js';
+import {getPlayerCrouchFactor} from './character-controller.js';
+import ioManager from './io-manager.js';
+import metaversefile from 'metaversefile';
+
+const localVector = new THREE.Vector3();
+const localVector2 = new THREE.Vector3();
+const localVector3 = new THREE.Vector3();
+const localVector4 = new THREE.Vector3();
+const localVector5 = new THREE.Vector3();
+const localQuaternion = new THREE.Quaternion();
+const localQuaternion2 = new THREE.Quaternion();
+const localMatrix = new THREE.Matrix4();
+
+
+
+class CharacterSfx {
+  constructor(player) {
+    this.player = player;
+
+    this.shouldExpectLeftStep = false;
+    this.shouldExpectRightStep = false;
+
+    this.prevLeftZ = 10;
+    this.prevRightZ = 10;
+  }
+
+  round2(num) {
+    return +(Math.round(num + "e+2")  + "e-2");
+  }
+  
+  updateSteps(now, timeDiffS) {
+    
+    if (this.player === null || this.player.avatar === null) {
+      return;
+    }
+
+    //debugger;
+
+    const soundManager = metaversefile.useSoundManager();
+
+    const boneFootLeft = this.player.avatar.foundModelBones['Left_ankle'];
+    const boneFootRight = this.player.avatar.foundModelBones['Right_ankle'];
+
+    let posLeft = 0;
+    let posRight = 0;
+
+    //console.log(boneFootLeft);
+
+    boneFootLeft.matrix.copy(boneFootLeft.matrixWorld).
+        premultiply(localMatrix.copy(this.player.avatar.modelBoneOutputs.Root.matrixWorld).invert());
+
+    boneFootLeft.matrix.decompose(localVector, localQuaternion, localVector2);
+
+    posLeft = localVector.z;

```

Comment:
Why z? Shouldn't it be Y?

Either 1) this is a mistake, or 2) this is compensating for some weird rotation axis switch, which we should be multiplying out to do correct math, rather than switching axis here as a hack.